README.rst

```diff
@@ -235,3 +237,21 @@ debugging information before your first request:
         print("Handling first request at: {}".format(now))
 
     server.register_before_first_request(print_time)
+
+Accessing the HTTP request from JSON-RPC methods
+------------------------------------------------
+In some situations, you may want to access the HTTP request from your JSON-RPC method. A typical
+example would be the need to perform logic based on headers in the request. In the ``server``
+module, there is a special ``CURRENT_REQUEST`` attribute which allows you to access the HTTP request
+which was used to call the current method. It is implemented as a thread-local, so you cannot make
+calls back into ``Server`` from a JSON-RPC method.
```

Comment:
This could use a bit more information about what exactly you cannot do (maybe just say that calling the dispatch method would overwrite `CURRENT_REQUEST` because it doesn't create a new thread).